### DISCUSSION TITLE
GA4 availability in Redash?

### DISCUSSION THREAD

**Discussion by M-arghandiwal** [getredash/redash#6645] [Q&A] [answered]
I can't find anything that instructs me on how to get data from Google Analytics 4. Is it possible or not? Or I am missing something

**Comment by masayuki038** [comment]
Hi @M-arghandiwal
From https://github.com/getredash/redash/issues/6097, GA4 is supported on master branch and `redash/redash:preview` container.

**Reply by M-arghandiwal** [reply]
Thank you @masayuki038 
I had already connected my Google Analytics with Redash, I got a request JSON from GA4 Query Explorer, but when I give it to Redash, it gives me "Error running query: Got an unexpected keyword argument "dateRanges""

The Test Connection is giving Success, 
Redash version is 10.1.0

**Reply by masayuki038** [reply, marked as answer]
@M-arghandiwal Yes, [the PR](https://github.com/getredash/redash/pull/5868) was merged in July this year. So, GA4 isn't supported in Redash 10.1.0. For now, you can use [redash/redash:preview](https://hub.docker.com/layers/redash/redash/preview/images/sha256-a5a0c4ddda60fe65fb186c868347cc2171e00d18bbde3de78d19108156888d1e) container or build master branch by yourself or wait for the next release

**Reply by M-arghandiwal** [reply]
Thank You! @masayuki038
